lib/cpp/test/CMakeLists.txt

```diff
@@ -25,6 +25,10 @@ include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
 #Make sure gen-cpp files can be included
 include_directories("${CMAKE_CURRENT_BINARY_DIR}")
 
+if (MINGW)
+  add_definitions(-DBOOST_TEST_DYN_LINK)
+endif()
+
```

Comment:
I don't think `BOOST_TEST_DYN_LINK` is related to mingw.
You may want to pass it to cmake by `-DCMAKE_CXX_FLAGS="-D..."` option instead.